exporter/views.py

```diff
@@ -33,7 +41,7 @@ def get_original_asset_id(download_url):
     that identifies this image uniquely on loc.gov
     """
     if download_url.startswith("http://tile.loc.gov/"):
-        pattern = r"/service:([A-Za-z0-9:]*)/"
+        pattern = r"/service:([A-Za-z0-9:\-]*)/"
```

Comment:
Looking at the code for the other side of this, I see this comment:

> The API places no restrictions on the form of the identifiers that a server may use or support. All special characters (e.g. ? or #) must be URI encoded to avoid unpredictable client behaviors. The URI syntax relies upon slash (/) separators so any slashes in the identifier must be URI encoded (also called "percent encoded").

That is basically passed in to this code:

```python
    identifier = urllib.unquote_plus(raw_id).replace(':', '/')
```

Since that's using [Flask's `path` pattern](http://exploreflask.com/en/latest/views.html#built-in-converters) I'm interpreting that as potentially accepting any character which a POSIX filesystem will accept.

Reply:
Do you think `pattern = r"/service:([^/]?)/" ` would be too permissive?

Reply:
Should that be `r"/service:([^/]+)/"` to require one or more non-path separators? It seems reasonable to me but the main question I would have is how we'd know if that belief is incorrect — since this isn't in a public blocking path the big thing I'd focus on is getting a clear error report.